.tests/tests/jupytertest.spec.ts

```diff
@@ -0,0 +1,125 @@
+    if(idx >= files.length) {
+        break;
+    }
+
+    const name = files[idx];
+    if(name === '') {
+        continue;
+    }
+    test(`test ${i}: ${name}`, async ({page},testInfo) => {
+        test.setTimeout(10800000);
+
+        // Create Env or get existing Env
+        const env = await envPool.getEnv(testInfo.parallelIndex);
+        
+        await sleep(5000);
+        console.log('url:'+env.getJuypterUrl(name.substring(3)));
+        
+        await page.goto(env.getJuypterUrl(name.substring(3)));
+
+        await page.waitForLoadState(); 
+
+        // Wait for content to appear before moving on.  Jupyter is slow to load (Done)  (Hack job)
+        await page.locator('span[class="f1235lqo"]').getByText('| Idle').waitFor({timeout: 300000})
+
+        const data = JSON.parse(fs.readFileSync(name, 'utf8'));
+        console.log('data: ' + data.metadata.kernelspec.language);
+
+        let strKernelType = '';
+        if (data.metadata.kernelspec.language === 'python') {
+            console.log('Python Notebook');
+            strKernelType = "Python 3 (ipykernel) ";
+        } 
+        else if (data.metadata.kernelspec.language === 'Teradata SQL') {
+            console.log('Teradata SQL Notebook');
+            strKernelType = 'Teradata SQL ';
+        } else if (data.metadata.kernelspec.language === 'R') {
+            console.log('R Notebook');
+            strKernelType = 'R ';
+        }
+        
+        console.log('Kernel Type: ' + strKernelType);   
+        expect(strKernelType).not.toBe('');
+        await page.locator('span[class="f1235lqo"] >> text="'+strKernelType+'| Idle"').waitFor({timeout: 600000});      
+
+        //Get the number of cells in the notebook
+        const jpCells =  await page.locator('div.jp-NotebookPanel:not(.p-mod-hidden)> div > div.jp-Cell');
+        await jpCells.first().waitFor({timeout: 300000});
+        console.log('jpCells: ' + await jpCells.count());
+        
+        var dm = await jpCells.count(); // Default Number of iterations for each Notebook demo
+
+        jpCells.first().click();
+        for (let i = 0; i < dm; i++) {
+            await sleep(100);
+            expect(await page.locator(".jp-RenderedText[data-mime-type='application/vnd.jupyter.stderr']")).toHaveCount(0);
+            await page.locator('span[class="f1235lqo"] >> text="'+strKernelType+'| Idle"').waitFor({timeout: 600000});      
+            if (await page.locator('span[class="f1235lqo"] >> text="'+strKernelType+'| Idle"').isVisible())
+            {
+                // Go to next step by clicking Shift+Enter
+                await page.getByRole('button', { name: 'Run the selected cells and' }).click();
+                
+            }    
+            try{
+                await page.locator('span[class="f1235lqo"] >> text="'+strKernelType+'| Busy"').waitFor({timeout: 2000});
+                expect
+            } catch (e) {
+                continue;
```

Comment:
Are we not swallowing an exception here?

Reply:
Ya, That's the idea. Basically what happens is for the txt only steps there is no execution or wait required. So this block checks whether the Busy state change happens that indicate there is some processing step. And didn't find any elegant way of doing this. The playwright considers timeout as exception. But as for us the timeout is expected.

Reply:
Ok, then perhaps the code should check for the exception message and rethrow everything that is not a timeout?